src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -116,6 +116,59 @@ private bool IsConvenienceMethodMeaningful()
                 || !_returnType.Convenience.Equals(_returnType.Protocol);
         }
 
+        private bool IsLowConfident()
+        {
+            var visitedModels = new HashSet<InputModelType>();
```

Comment:
I'm thinking this method do once per method. But models are shared across library. So there will be some redundant calculation.

Reply:
Yes there is duplicated calculation.
Now the IsLowConfident flag does not propagate. I will see if we could make it propagates on the emitter so that we could just read them here.

Reply:
I changed the implementation to calculation this in the emitter so that there should be no more duplicated calculations